src/utils.ts

```diff
@@ -520,4 +522,30 @@ const katexMarkedExtension = {
     }),
 };
 
+export const renderer = {
+  // Add escaped header ids for anchor hash browsing
+  heading(text: string, level: 1 | 2 | 3 | 4 | 5 | 6) {
+    const escapedText = text
+      .toLowerCase()
+      .replace(/[^\w]+/g, "-")
+      .replace(/^-|-$/g, "");
+
+    return `<h${level} id="${escapedText}">${text}</h${level}>`;
+  },
+};
```

Comment:
Can you document the purpose of this function?

Reply:
yeah sure. np

Reply:
6cef9f10